Pod/Classes/SwitchViewComponent/ARSwitchViewManager.m

```diff
@@ -2,33 +2,47 @@
 
 #import <Extraction/ARSwitchView.h>
 #import <React/RCTComponent.h>
+#import <FLKAutoLayout/FLKAutoLayout.h>
 
 
-@interface ARSwitchViewComponent : ARSwitchView <ARSwitchViewDelegate>
+@interface ARSwitchViewComponent : UIView <ARSwitchViewDelegate>
+@property (nonatomic, strong, readwrite) ARSwitchView *switchView;
 @property (nonatomic, strong, readwrite) RCTDirectEventBlock onSelectionChange;
 @end
 
 @implementation ARSwitchViewComponent
 
-- (instancetype)initWithButtonTitles:(NSArray *)titles;
+- (instancetype)init;
 {
-  if ((self = [super initWithButtonTitles:titles])) {
-    self.delegate = self;
-  }
-  return self;
+    if ((self = [super init])) {
+        
+        // We are wrapping ARSwitchView within a UIView.
+        // This is done to intercept React Native re-setting the background color on ARSwitchView directly.
```

Comment:
For anyone following along in this issue, this is what happened (cc @orta):

If no background color style is set in a React Native component, it will inherit the background color from its parent component (as happens in web). 

This inherited color will be the final color. 

If you have a background color property set on your view in your native code, it will be overriden, as the RN code will be called _after_. 

What we’ve chosen to do (thanks @alloy!) is have the bridging component inherit from `UIView` (as we don’t care about it’s background color) rather than `ARSwitchView`, hence the switch view will keep its proper colors.